demisto_sdk/commands/common/hook_validations/field_base_validator.py

```diff
@@ -362,3 +366,67 @@ def does_not_have_empty_select_values(self) -> bool:
                                  warning=self.structure_validator.quite_bc):
                 return False
         return True
+
+    def is_aliased_fields_are_valid(self) -> bool:
+        """
+        Validates that the aliased fields (fields that appear as Aliases in another field) are valid.
+        invalid aliased fields are
+        1. fields that are in another field's Aliases list, and present in the same marketplace of that field.
+        2. fields that are in another field's Aliases list, and also contain Aliases (nested aliasing)
+
+        Returns:
+            (bool): True if aliased fields are valid.
+        """
+
+        if not self.id_set_file:
+            print_warning('Validate will skip since an id set file was not provided')
+            return True
+
+        aliases = self.current_file.get('Aliases', [])
+        if not aliases:
+            return True
+
+        is_valid = True
+        validators_and_error_generators = [
+            (self.is_alias_has_invalid_marketplaces, Errors.invalid_marketplaces_in_alias),
+            (self.is_alias_has_inner_alias, Errors.aliases_with_inner_alias),
+        ]
+        aliased_fields = self._get_incident_fields_by_aliases(aliases)
+        for validator, error_generator in validators_and_error_generators:
+            invalid_aliased = [alias.get("cliName") for alias in aliased_fields if validator(alias)]
+            if invalid_aliased:
+                error_message, error_code = error_generator(invalid_aliased)
+                if self.handle_error(error_message, error_code, file_path=self.file_path, warning=self.structure_validator.quite_bc):
+                    is_valid = False
+
+        return is_valid
+
+    def _get_incident_fields_by_aliases(self, aliases: List[dict]) -> List[dict]:
+        """Get from the id_set the actual fields for the given aliases
+
+        Args:
+            aliases (list): The alias list.
+
+        Returns:
+            (list): the fileds as list of dicts.
+        """
+        res = []
+        alias_ids: set = {f'incident_{alias.get("cliName")}' for alias in aliases}
+        incident_field_list: list = self.id_set_file.get('IncidentFields')
+
+        for incident_field in incident_field_list:
```

Comment:
Same here, use `yield`.

Reply:
regarding the need to open the file - i didn't see the info about aliases of the incident field in the id set
am i miss something ?

Reply:
It was recently added: `aliases: List[str] = sum(([field['cliName'], field['name']] for field in json_data.get('Aliases', [])), [])`.
From where did you get the ID set?